node/bft/ledger-service/src/traits.rs

```diff
@@ -118,4 +118,8 @@ pub trait LedgerService<N: Network>: Debug + Send + Sync {
     /// Adds the given block as the next block in the ledger.
     #[cfg(feature = "ledger-write")]
     fn advance_to_next_block(&self, block: &Block<N>) -> Result<()>;
+
+    // Updates the block metrics.
+    #[cfg(feature = "metrics")]
+    fn update_block_metrics(&self, _block: &Block<N>) {}
```

Comment:
Just as a general nit, traits are interfaces and when other Rust code needs to accept a trait parameter elsewhere, it expects the concrete type that implements it to have it implemented as well. 

Putting a feature flag in a trait basically makes other implementors of that trait need to also implement a feature flag in their implementation further down the line which can be lead to a lot of problems. Thus in general it's good NOT to put feature flags in traits for this reason unless it's somehow absolutely necessary. (There is another feature flag in the trait above, but this is now adding a separate one) 

In the concrete implementation of this above, the &self parameter isn't used and only block data from the parameter `block` input itself is ever used (i.e. you don't need to access &self at all). Therefore you don't need it as part of the trait and can make the `update_block_metrics` call from either a standalone function or a method on another object.

Luckily though this should be a simple change!

Reply:
Yeah I was initially trying to make it a standalone function but for some reason it would not compile. Trying this again now

Reply:
Changed :D